tools/ivwpy/regression/generatereport.py

```diff
@@ -450,10 +450,21 @@ def images(self, imgs, testdir):
         def path(type, img):
             return os.path.relpath(toPath(testdir, type, img), self.basedir)
 
-        def imgstatus(key):
+        def imgstatus(img, warnPercentage=10):
+            # Return the status of the image. Either "fail", "ok", or "warning".
+            # Warn if within warnPercentage % of the difference tolerance
             failures = safeget(self.report, 'failures', failure={})
-            fail = next((x for x in failures if x[0] == "images"), None)
-            return "ok" if fail is None else "fail"
+            failedImgs = next((x for x in failures if x[0] == "images"), None)
+            failed = (failedImgs is not None and img["image"] in failedImgs[1])
+            diffTolerance = safeget(self.report, "config", "image_test", "differenceTolerance", 
+                                    img["image"], failure=0.0)
+
+            if failed:
+                return "fail"
+            elif img["difference"] > diffTolerance * (1.0 - warnPercentage/100.0):
```

Comment:
still not sure how useful this is, differenceTolerance is almost always 0...

Reply:
Agreed. If it is 0 most of the time, then this is hardly useful.

Reply:
The default is 0, and it is only if we manually change the value in the config file for that test that it will be nonzero. And I don't think we do that very often...

Reply:
the differences does not work very well since even a tiny visual change can have a huge difference. We would need some kind of visual similarity here for the tolerance to be useful. But on the other hand then we might miss a lot of regressions that only change minor aspects (like text moving halv a pixel), but that we still want to catch...